src/lib/web-component/log-view/log-view-element.js

```diff
@@ -0,0 +1,85 @@
+class LogView extends HTMLElement {
+  constructor() {
+    super();
+
+    const shadow = this.attachShadow({ mode: 'open' });
+
+    this.logTableTemplate = document.querySelector('#log-table-template');
+    this.logTableRowTemplate = document.querySelector(
+      '.log-table-row-template'
+    );
+
+    const logTableInstance = this.logTableTemplate.content.cloneNode(true);
+    this.logDetailWrapper = logTableInstance.querySelector(
+      '.log-detail-wrapper'
+    );
+    this.logDetails = logTableInstance.querySelector('.log-details');
+    this.closeBtn = logTableInstance.querySelector('.close');
+    this.logTableWrapper = logTableInstance.querySelector('.log-table-wrapper');
+    this.tableBody = logTableInstance.querySelector('tbody');
+
+    shadow.appendChild(logTableInstance);
+  }
+
+  addNewRows(logs) {
+    const rowsFragment = document.createDocumentFragment();
+
+    for (const log of logs) {
+      const logTableRowInstance = this.logTableRowTemplate.content.cloneNode(
```

Comment:
I suggest to store the firstElementChild in th constructor, so you don't have to clone the superfluous whitespace. This code is relatively hot (used often), so it'll be good to minimize the amount of redundant calls.

Furthermore, by storing it in the constructor, it will be more obvious that there is only one (root) node in the template.